src/types/redis_hash.cc

```diff
@@ -418,4 +472,208 @@ rocksdb::Status Hash::RandField(const Slice &user_key, int64_t command_count, st
   return rocksdb::Status::OK();
 }
 
+rocksdb::Status Hash::ExpireFields(const Slice &user_key, uint64_t expire_ms, const std::vector<Slice> &fields,
+                                   HashFieldExpireType type, bool is_persist, std::vector<int8_t> *ret) {
+  std::string ns_key = AppendNamespacePrefix(user_key);
```

Comment:
Need to add the lock guard to prevent data race for the user key.

Reply:
done